sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c

```diff
@@ -100,6 +97,54 @@ kernel_pid_t gnrc_ipv6_init(void)
     return gnrc_ipv6_pid;
 }
 
+static bool _dispatch_next_header(gnrc_pktsnip_t *current,
+                                  gnrc_pktsnip_t *pkt,
+                                  uint8_t nh,
+                                  bool interested)
+{
+#ifdef MODULE_GNRC_IPV6_EXT
+    const bool current_is_ipv6 = ((current->type != GNRC_NETTYPE_IPV6_EXT) ||
+                                  (current->next->type == GNRC_NETTYPE_IPV6));
+#else
+    const bool current_is_ipv6 = (current->next->type == GNRC_NETTYPE_IPV6);
```

Comment:
`current_is_ipv6` doesn't reflect what it is. `should_dispatch_current_type` or something would be better.